myhoard/controller.py

```diff
@@ -1304,8 +1334,15 @@ def _purge_old_backups(self):
             self.log.warning("Backup %r to drop is one of active streams, not dropping", backup["stream_id"])
             return
 
-        self._build_backup_stream(backup).remove()
+        # lock the controller, this way other requests do not access backups till backup is purged
         with self.lock:
+            self._build_backup_stream(backup).remove()
+            current_backups = [
+                current_backup
+                for current_backup in self.state["backups"]
+                if current_backup["stream_id"] != backup["stream_id"]
+            ]
+            self.state_manager.update_state(backups=current_backups)
```

Comment:
Updated list of backups here, otherwise the purged backup might still be listed (_refresh_backup_list might skip refreshing in the next iteration)